lib/dat-manager.js

```diff
@@ -0,0 +1,138 @@
+const collect = require('collect-stream')
+const Stats = require('hyperdrive-stats')
+const sub = require('subleveldown')
+const db = require('./db')
+const encoding = require('dat-encoding')
+const Dat = require('dat-js')
+const root = require('./root-dir')
+const fs = require('fs')
+const EventEmitter = require('events')
+
+module.exports = class Manager extends EventEmitter {
+  constructor () {
+    super()
+    this.dats = []
+    this.loadAll((err, _dats) => {
+      if (err) return this.emit('error', err)
+      for (let dat of _dats) this.dats.push(this.watch(dat))
+      this.emit('update')
+    })
+  }
+  watch (dat) {
+    dat.on('update', () => this.emit('update'))
+    return dat
+  }
+  get () {
+    return this.dats
+  }
+  create (dir, cb) {
+    const dat = Dat({ dir })
+    dat.owner = true
+    dat.dir = dir
+    dat.share(err => {
+      if (err) return cb(err)
+      this.persist(dat, err => {
+        if (err) return cb(err)
+        this.dats.push(this.watch(this.activate(dat)))
+        this.emit('update')
+        cb()
+      })
+    })
+  }
+  download (key, cb) {
+    const dir = `${root}/${key}`
+    fs.mkdir(dir, err => {
+      if (err) return cb(err)
+      const dat = Dat({ dir, key })
+      dat.owner = false
+      dat.dir = dir
+      this.persist(dat, err => {
+        if (err) return cb(err)
+        dat.download()
+        this.dats.push(this.watch(this.activate(dat)))
+        this.emit('update')
+        cb()
+      })
+    })
+  }
+  load ({ key, owner, dir }) {
+    if (Buffer.isBuffer(key)) key = encoding.encode(key)
+    const dat = Dat({ key, dir })
+    dat.dir = dir
+    dat.owner = owner
+    if (dat.owner) {
+      dat.share(err => {
+        if (err) dat.emit('error', err)
+      })
+    } else {
+      dat.download(err => {
```

Comment:
What if we put these share/download calls into activate? Then load doesn't start any processes